File: scripts/process_new_avatars.js
const axios = require('axios')
const fs = require('fs')
const path = require('path')
const https = require('https')
const FormData = require('form-data')

const API_KEY = 'fzotK6cmX6mEkRyr5ma8aiFy'
const SOURCE_DIR = '/Users/macmini/Documents/Primary_Images'
const OUT_DIR = '/Users/macmini/south-cinema-analytics/frontend/public/avatars'
const WORKTREE_OUT_DIR = '/Users/macmini/south-cinema-analytics/.claude/worktrees/relaxed-euler/frontend/public/avatars'

const httpsAgent = new https.Agent({ rejectUnauthorized: false })

// Map: source filename (lowercase, no ext) → target slug (matches existing site convention)
const NEW_ACTORS = {
  'arjunsarja':          'arjunsarja',
  'chiranjeevi':         'chiranjeevi',
  'nagarjunaakkineni':   'nagarjunaakkineni',
  'nandamuribalakrishna':'nandamuribalakrishna',
  'ranadaggubati':       'ranadaggubati',
  'raviteja':            'raviteja',
  'sharwanand':          'sharwanand',
  'siddhujonnalagadda':  'siddhujonnalagadda',
  'varunsandesh':        'varunsandesh',
  'varuntej':            'varuntej',
  'venkateshdaggubati':  'venkateshdaggubati',
  'vishwaksen':          'vishwaksen',
}

async function processActor(sourceSlug, targetSlug) {
  // Find source file (case-insensitive)
  const files = fs.readdirSync(SOURCE_DIR)
  const match = files.find(f => f.replace(/\.png$/i, '').toLowerCase() === sourceSlug.toLowerCase())
  if (!match) {
    console.log(`✗ ${sourceSlug}: source file not found`)
    return false
  }

  const sourcePath = path.join(SOURCE_DIR, match)
  const outPath    = path.join(OUT_DIR, `${targetSlug}.png`)
  const worktreePath = path.join(WORKTREE_OUT_DIR, `${targetSlug}.png`)

  console.log(`⏳ ${targetSlug}: sending to remove.bg...`)

  try {
    const form = new FormData()
    form.append('image_file', fs.createReadStream(sourcePath))
    form.append('size', 'auto')

    const response = await axios.post('https://api.remove.bg/v1.0/removebg', form, {
      headers: { ...form.getHeaders(), 'X-Api-Key': API_KEY },
      responseType: 'arraybuffer',
      httpsAgent,
      timeout: 30000,
    })

    fs.writeFileSync(outPath, response.data)
    fs.writeFileSync(worktreePath, response.data)
    console.log(`✓ ${targetSlug}: saved (${Math.round(response.data.length / 1024)}KB)`)
    return true
  } catch (err) {
    const msg = err.response
      ? `HTTP ${err.response.status}: ${Buffer.from(err.response.data).toString().slice(0, 120)}`
      : err.message
    console.log(`✗ ${targetSlug}: ${msg}`)
    return false
  }
}

async function main() {
  const entries = Object.entries(NEW_ACTORS)
  console.log(`Processing ${entries.length} new actors...\n`)
  let ok = 0, fail = 0
  for (const [src, tgt] of entries) {
    const success = await processActor(src, tgt)
    success ? ok++ : fail++
    await new Promise(r => setTimeout(r, 600)) // gentle rate limit
  }
  console.log(`\nDone — ${ok} succeeded, ${fail} failed`)
}

main()

File: scripts/process_new_avatars.py
import os, ssl, time, shutil
import urllib.request
import urllib.parse

API_KEY    = 'fzotK6cmX6mEkRyr5ma8aiFy'
SOURCE_DIR = '/Users/macmini/Documents/Primary_Images'
OUT_DIR    = '/Users/macmini/south-cinema-analytics/frontend/public/avatars'
WORKTREE   = '/Users/macmini/south-cinema-analytics/.claude/worktrees/relaxed-euler/frontend/public/avatars'

# source filename slug → target avatar slug
NEW_ACTORS = {
    'arjunsarja':           'arjunsarja',
    'chiranjeevi':          'chiranjeevi',
    'nagarjunaakkineni':    'nagarjunaakkineni',
    'nandamuribalakrishna': 'nandamuribalakrishna',
    'ranadaggubati':        'ranadaggubati',
    'raviteja':             'raviteja',
    'sharwanand':           'sharwanand',
    'siddhujonnalagadda':   'siddhujonnalagadda',
    'varunsandesh':         'varunsandesh',
    'varuntej':             'varuntej',
    'venkateshdaggubati':   'venkateshdaggubati',
    'vishwaksen':           'vishwaksen',
}

ctx = ssl.create_default_context()
ctx.check_hostname = False
ctx.verify_mode = ssl.CERT_NONE

def find_source(slug):
    for f in os.listdir(SOURCE_DIR):
        if f.lower().replace('.png','') == slug.lower():
            return os.path.join(SOURCE_DIR, f)
    return None

def process(src_slug, tgt_slug):
    src = find_source(src_slug)
    if not src:
        print(f'✗ {tgt_slug}: source file not found')
        return False

    with open(src, 'rb') as f:
        img_data = f.read()

    boundary = 'SCABoundary12345'
    body = (
        f'--{boundary}\r\n'
        f'Content-Disposition: form-data; name="size"\r\n\r\nauto\r\n'
        f'--{boundary}\r\n'
        f'Content-Disposition: form-data; name="image_file"; filename="image.png"\r\n'
        f'Content-Type: image/png\r\n\r\n'
    ).encode() + img_data + f'\r\n--{boundary}--\r\n'.encode()

    req = urllib.request.Request(
        'https://api.remove.bg/v1.0/removebg',
        data=body,
        headers={
            'X-Api-Key': API_KEY,
            'Content-Type': f'multipart/form-data; boundary={boundary}',
        },
        method='POST'
    )

    print(f'⏳ {tgt_slug}: processing...')
    try:
        with urllib.request.urlopen(req, timeout=30, context=ctx) as resp:
            result = resp.read()
            out = os.path.join(OUT_DIR, f'{tgt_slug}.png')
            wt  = os.path.join(WORKTREE, f'{tgt_slug}.png')
            with open(out, 'wb') as f: f.write(result)
            shutil.copy(out, wt)
            print(f'✓ {tgt_slug}: saved ({len(result)//1024}KB)')
            return True
    except urllib.error.HTTPError as e:
        print(f'✗ {tgt_slug}: HTTP {e.code} — {e.read().decode()[:120]}')
        return False
    except Exception as e:
        print(f'✗ {tgt_slug}: {e}')
        return False

ok = fail = 0
print(f'Processing {len(NEW_ACTORS)} new actors...\n')
for src, tgt in NEW_ACTORS.items():
    if process(src, tgt):
        ok += 1
    else:
        fail += 1
    time.sleep(0.6)

print(f'\nDone — {ok} succeeded, {fail} failed')
